api/v1alpha8/types.go

```diff
@@ -357,6 +363,11 @@ type APIServerLoadBalancer struct {
 	Provider string `json:"provider,omitempty"`
 }
 
+// ReferencedMachineResources contains resolved references to resources required by a machine.
```

Comment:
I wish we added more doc to this one. We need to clarify why we have that, how devs can use it for future resources etc.

Reply:
Would it make sense to drag this whole feature out to a separate PR?

Reply:
@stephenfin: I don't see a lot of merit in introducing an empty field in a separate PR. We can discuss it's sense here, and circle back to putting ServerGroupID back onto the MachineStatus if we decide it's better.

Reply:
I added some docs.